pkg/query-service/app/querier/v2/helper.go

```diff
@@ -276,3 +277,59 @@ func (q *querier) runBuilderQuery(
 		Series: resultSeries,
 	}
 }
+
+// ValidateMetricNames function is used to print all those queries who are still using old normalized metrics and not new metrics.
+func (q *querier) ValidateMetricNames(ctx context.Context, query *v3.CompositeQuery, orgID valuer.UUID) {
+	var metricNames []string
+	switch query.QueryType {
+	case v3.QueryTypePromQL:
+		for _, query := range query.PromQueries {
+			expr, err := parser.ParseExpr(query.Query)
+			if err != nil {
+				zap.L().Debug("error parsing promQL expression", zap.String("query", query.Query), zap.Error(err))
+				continue
+			}
+			parser.Inspect(expr, func(node parser.Node, path []parser.Node) error {
+				if vs, ok := node.(*parser.VectorSelector); ok {
+					for _, m := range vs.LabelMatchers {
+						if m.Name == "__name__" {
+							metricNames = append(metricNames, m.Value)
+						}
+					}
+				}
+				return nil
+			})
+		}
+		metrics, err := q.reader.GetNormalizedStatus(ctx, orgID, metricNames)
+		if err != nil {
+			zap.L().Debug("error getting corresponding normalized metrics", zap.Error(err))
+			return
+		}
+		for k, m := range metrics {
+			if m {
+				continue
+			} else {
+				zap.L().Warn("using normalized metric name", zap.String("metrics", k))
+				continue
+			}
+		}
+	case v3.QueryTypeBuilder:
+		for _, query := range query.BuilderQueries {
+			metricName := query.AggregateAttribute.Key
+			metricNames = append(metricNames, metricName)
+		}
+		metrics, err := q.reader.GetNormalizedStatus(ctx, orgID, metricNames)
+		if err != nil {
+			zap.L().Debug("error getting corresponding normalized metrics", zap.Error(err))
+			return
+		}
+		for k, m := range metrics {
+			if m {
+				continue
+			} else {
+				zap.L().Warn("using normalized metric name", zap.String("metrics", k))
+				continue
+			}
+		}
```

Comment:
Please use readable names, what should the reader interpret as just reading the `k, m`?